pineappl/src/grid.rs

```diff
@@ -1815,6 +1779,59 @@ impl Grid {
         FkTable::try_from(result).ok()
     }
 
+    /// Converts this `Grid` into an [`FkTable`] using an evolution kernel operator (EKO) given as
+    /// `operator`. The dimensions and properties of this operator must be described using `info`.
+    /// The parameter `order_mask` can be used to include or exclude orders from this operation,
+    /// and must correspond to the ordering given by [`Grid::orders`]. Orders that are not given
+    /// are enabled, and in particular if `order_mask` is empty all orders are activated.
+    ///
+    /// # Errors
+    ///
+    /// Returns a [`GridError::EvolutionFailure`] if either the `operator` or its `info` is
+    /// incompatible with this `Grid`.
+    pub fn evolve(
+        &self,
+        operator: &Array5<f64>,
+        info: &OperatorInfo,
+        order_mask: &[bool],
+    ) -> Result<FkTable, GridError> {
+        let op_info_dim = (
+            info.fac1.len(),
+            info.pids1.len(),
+            info.x1.len(),
+            info.pids0.len(),
+            info.x0.len(),
+        );
+
+        if operator.dim() != op_info_dim {
+            return Err(GridError::EvolutionFailure(format!(
+                "operator information {:?} does not match the operator's dimensions: {:?}",
+                op_info_dim,
+                operator.dim(),
+            )));
+        }
+
+        let (subgrids, lumi) = if self.has_pdf1() && self.has_pdf2() {
+            evolution::evolve_with_two(self, operator, info, order_mask)
+        } else {
+            evolution::evolve_with_one(self, operator, info, order_mask)
+        }?;
+
+        let mut grid = Self {
+            subgrids,
+            lumi,
+            bin_limits: self.bin_limits.clone(),
+            orders: vec![Order::new(0, 0, 0, 0)],
+            subgrid_params: SubgridParams::default(),
+            more_members: self.more_members.clone(),
+        };
```

Comment:
That's correct!

Reply:
However, I think having this method there is perfectly fine, it's the entrypoint for evolution, does does check the validity of its arguments and constructs the final `FkTable`. The main work is done in the `evolution` module.

Reply:
That's fine, but I would also like to provide the full feature on that side.

However, it is just a different structure, and at the moment a problematic one. I would simply forget about it, at least for a while.

Reply:
This will definitely change in `v0.6.0`; in this new version I'll make some breaking changes (only in the Rust API) and many of the members will be modified or even removed.